Python/Product/Profiling/Profiling/ProfiledProcess.cs

```diff
@@ -30,11 +30,20 @@ sealed class ProfiledProcess : IDisposable {
         private readonly ProcessorArchitecture _arch;
         private readonly Process _process;
         private readonly PythonToolsService _pyService;
-
+        public string pyArch { get; set; }
         public ProfiledProcess(PythonToolsService pyService, string exe, string args, string dir, Dictionary<string, string> envVars) {
-            var arch = NativeMethods.GetBinaryType(exe);
-            if (arch != ProcessorArchitecture.X86 && arch != ProcessorArchitecture.Amd64) {
-                throw new InvalidOperationException(Strings.UnsupportedArchitecture.FormatUI(arch));
+
+            string pythonInstallDir = Path.GetDirectoryName(PythonToolsInstallPath.GetFile("VsPyProf.dll", typeof(ProfiledProcess).Assembly));
+
+            var arg = ProcessOutput.QuoteSingleArgument(Path.Combine(pythonInstallDir, "getArch.py"));
+            getPyArch(exe, arg);
+
+            if (pyArch == "64") {
+                _arch = ProcessorArchitecture.Amd64;
+            } else if (pyArch == "32") {
+                _arch = ProcessorArchitecture.X86;
+            } else {
+                throw new InvalidOperationException(Strings.UnsupportedArchitecture.FormatUI(_arch));
```

Comment:
This error message doesn't make sense. It's supposed to be printing the arch that is unsupported, but you're printing the value of `_arch`, which hasn't been assigned if this code path is taken.

I believe you want to print `pyArch` here instead.